packages/frontend-api/src/Routing/FrontendApiRouteLoader.php

```diff
@@ -39,7 +48,11 @@ public function loadRoutes(): RouteCollection
     {
         $routes = new RouteCollection();
 
-        if (!$this->enabledOnCurrentDomainChecker->isEnabledOnCurrentDomain()) {
+        try {
+            if (!$this->enabledOnDomainChecker->isEnabledOnDomain($this->domain->getId())) {
+                return $routes;
+            }
+        } catch (NoDomainSelectedException $e) {
```

Comment:
Where is this exception thrown?

Reply:
`$this->domain->getId()` throws this exception :)